chain/chunks/src/lib.rs

```diff
@@ -1075,6 +1089,136 @@ impl ShardsManager {
         )
     }
 
+    fn prepare_partial_encoded_chunk_response_from_chunk(
```

Comment:
It’s not immediately obvious to be me what should be moved to separate file though to be honest. Code that handles partial encoded chunk requests is less than 200 lines so moving just it wouldn’t really help reduce the size of lib.rs. This would need some more care and thought to figure out which of the methods in the struct belong together in a separate file.

Reply:
Yeah, I don't think *this* makes sense to move to a separate file, as it seems to be pretty entangled with the rest.

The easiest wins in terms of keeping files small and logic clean are usually around various small helpers. Things like `RequestPool` or `SealsManager` seem like better candidates for extraction.